oauth1/signer_interceptor.py

```diff
@@ -37,16 +37,11 @@ def request_function(*args, **kwargs):  # pragma: no cover
         request_function.__oauth__ = True
         return request_function
 
-def add_signing_layer(self, api_client, key_file, key_password, consumer_key):
+def add_signing_layer(api_client, key_file, key_password, consumer_key):
```

Comment:
@mickfGit @pqlMC I have the impression (I'm not 100% sure) this is a breaking change, and will require the client applications to be updated. Could we keep the old function signature but only document the new one?

Reply:
This isn't a breaking change.  `self` is only useful for classes, and these two functions are library functions that are declared outside of the class, and self is never even referenced in the functions.  The changes in this PR are cleaner.

Reply:
Thanks! I was afraid it was a breaking API change since this is not the same number of parameters we pass to `add_signing_layer`.
If my application using version 1.1.2 of the library still compiles after I upgrade to v1.1.3 (without any code update) then it's fine for me.

Reply:
Ah yes.  That's a good catch... Since these are publicly exposed functions, then technically this is a breaking change I believe.  The naming of the params are just really confusing.

Reply:
The use of self in a function outside a class is useless, using v1.1.2 requires the user to add an unused trailing param that won't be used. We shouldn't keep a function that is technically wrong. Is there any way to retire/discourage the use of 1.1.2?
In any case, as a user, I would rather prefer to see a fixed function than have to deal with 2 identical functions, 1 with self and 1 without, as it would be confusing (why there are 2 functions? what's the difference between one and the other?).
This is the case of a library in its early stage, when a change like this is no harm.

Reply:
Yeah, I understand that the convention is useless outside a class but the param is still there.  `self` isn't a reserved keyword in Python.  If it were a private method encapsulated by the class where the number of params was being trimmed or expanded, then a minor bump would be fine.  However, since it's a publically exposed method directly invocable through the module with no encapsulation whatsoever, it's a breaking change.  Using the same function in 1.1.3 from 1.1.2 would not work as expected for all use cases.  I think it warrants a 2.0.0 version bump.

Reply:
I have updated the pull request with new changes:
- `def add_signer_layer(api_client, key_file, key_password, consumer_key)` has been added
- `def add_signing_layer(self, api_client, key_file, key_password, consumer_key)` has been marked as deprecated, the user will get an alert if he uses it